c7n/resources/awslambda.py

```diff
@@ -255,6 +255,58 @@ class KmsFilter(KmsRelatedFilter):
     RelatedIdsExpression = 'KMSKeyArn'
 
 
+@AWSLambda.action_registry.register('set-xray-tracing')
+class LambdaEnableXrayTracing(Action):
+    """
+        This action allows for enable Xray tracing to Active
+       :example:
+       .. code-block:: yaml
+           actions:
+             - type: enable-xray-tracing
+    """
+
+    schema = type_schema(
+        'set-xray-tracing',
+        **{'state': {'default': True, 'type': 'boolean'}}
+    )
+    permissions = ("lambda:UpdateFunctionConfiguration",)
+
+    def get_mode_val(self, state):
+        if state:
+            return "Active"
+        return "PassThrough"
+
+    def process(self, resources):
+        """
+            Enables the Xray Tracing for the function.
+
+            Args:
+                resources: AWS lamdba resources
+            Returns:
+                None
+        """
+        client = local_session(self.manager.session_factory).client('lambda')
+        updateState = self.data.get('state', True)
+        retry = get_retry(('TooManyRequestsException', 'ResourceConflictException'))
+
+        for resource in resources:
+            state = bool(resource["TracingConfig"]["Mode"] == "Active")
+            if updateState != state:
+                mode = self.get_mode_val(updateState)
```

Comment:
we can move this outside of the for loop

Reply:
Correct me if I'm wrong.
Here, we are evaluating each resource's `TracingConfig Mode` and above code should be inside loop!

Reply:
updateState from line 289 is from the policy itself so it shouldn't be changing from each resource